### DISCUSSION TITLE
Cypress is storing the same cookie with same sub domain

### DISCUSSION THREAD

**Discussion by GoushiRam** [cypress-io/cypress#27969] [Questions and Help]
With Cypress 12.5.1, we noticed that during set-cookie, the 2 cookies with the same name is added. Ideally the existing cookie should be replaced. This is not happening as the domain is added different for both the cookie locations. :
![image](https://github.com/cypress-io/cypress/assets/106053688/9841d485-cd0f-4567-8156-bcf9991234a2)

paf cookie being available in accounts.asgardeo.io and .accounts.asgardeo.io

This is observed only in cypress initiated browsers. What could be the reason for this?

**Comment by portyu9** [comment]
Hi @GoushiRam 

This was a Cypress bug, and your version (`12.5.1`) predates the fix.

Cypress `12.7.0` specifically fixed:

> cookies being duplicated with the same hostname, but a prepended dot

So a cookie shown for:

```text
accounts.asgardeo.io
```

and another for:

```text
.accounts.asgardeo.io
```

matches that known issue very closely.

The fix is to upgrade to at least Cypress `12.7.0` (preferably a currently supported Cypress version).

`12.7.0` also added explicit `hostOnly` handling to `cy.setCookie()`. Current Cypress lets you distinguish between a host-only cookie:

```js
cy.setCookie('name', 'value', {
  hostOnly: true,
})
```

and a domain cookie.

Also, according to the HTTP cookie specification, a leading dot in a `Domain` value is not supposed to create a different cookie scope in modern browsers; leading dots are effectively ignored. That is why seeing both forms duplicated was erroneous Cypress behavior rather than something you should work around in application code.

I could not retrieve the screenshot attached to the original discussion because the GitHub attachment URL is no longer accessible, but the version and textual description match the Cypress `12.7.0` changelog fix directly.

Please mark this answer as accepted if it helped, thank you.
